### PR TITLE
My_Calculator Milestone 5

### PR DESCRIPTION
**Pull request by Davidosky007**
# My_Calculator Milestone 5 
In this milestone
* App component was converted into a stateful parent component
* handleClick function was implemented in buttonPanel and button components
* total or next values were passed to the Display components to display current values
* handleClick function was passed as a clickHandler property to buttonPanel.

### CODE REVIEW

File: src/components/ButtonPanel.js [hunk 1]
```diff
@@ -1,39 +1,33 @@
...
-      <Button buttons="+" />
-    </div>
-    <div>
-      <Button buttons="0" />
-      <Button buttons="." />
-      <Button buttons="=" />
-    </div>
-  </>
+const buttonGroups = [];
+buttonGroups.push(['AC', '+/-', '%', '÷']);
```
Comment (Forison) [inline src/components/ButtonPanel.js]: **[Optional]**
I don't see the need of pushing into the `buttonGroups array` why not define it directly as a dimensional array, that way you don't have to perform a task of pushing.

File: src/components/App.js [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
-    <Display />
-    <ButtonPanel onClick={calculate} />
-  </>
-);
+import Calculate from '../logic/calculate';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
```
Comment (Forison) [inline src/components/App.js:10]: **[Optional]**
You could use arrow function in the function definition of `handleClick` to save your self from binding it in the constructor.

**Review by Forison — CHANGES_REQUESTED** [review]
## Status::Change required❌

# 👋  @Davidosky007 

#### You have done a good job on this one so far, You modified the app.js to class component and implemented the handleClick as required, however, there a functional flaw we have to address to get the project approved😇 .

* [ ] Let us handle the division of a number by zero well, it should return `undefined`. Currently the application throws an error when we try this computation.
* [ ] Let us address the error in the console of the browser.

#### That is it, kindly make the changes and submit a re-review🤗.

#### Happy coding😎

Addo Forison

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @Davidosky007   :clap: 

You still need to address the issue mentioned from the previous reveiwer;
In another word;
The app is broken once the user performs the following division operation by zero.
Division by zero is not handled properly yet. Let's fix this.
The reason is that when a division by zero happens, we should return an error message like `division error`

Divide a number by zero (e.g. 4 / 0)

# Status: Changes Required! ♻️ 

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**

- [ ] OPTIONAL - The display of the calculator buttons seems quite off. Consider arranging them with a few CSS styling to make usability easier.

Don't forget to fill the project completion form👍
Good luck with the next project.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

**Pull request merged**